File: server/skill_area/services/discipline.py
from skill_area.base import Service
from skill_area.models import StudentGroup, User, Discipline


class DisciplineService(Service):
    def create(self, _, name, group_id):
        discipline = Discipline(
            name=name,
            student_group_id=group_id
        )
        discipline.save()

        return {
            'id': discipline.id,
            'name': discipline.name,
            'group_id': discipline.student_group_id,
        }

    def get(self, _, id):
        discipline = Discipline.objects.get(id=id)

        return {
            'id': discipline.id,
            'name': discipline.name,
            'group_id': discipline.student_group_id,
        }

    def get_list(self, _, group_id=None):
        if group_id:
            disciplines = Discipline.objects.filter(student_group_id=group_id)
        else:
            disciplines = Discipline.objects.all()
        return [{
            'id': discipline.id,
            'name': discipline.name,
            'group_id': discipline.student_group_id,
        } for discipline in disciplines]

    def remove(self, id):
        discipline = Discipline.objects.get(id=id)
        discipline.delete()

File: client/src/services/discipline.js
import Service from '../base/Service.js'

class DisciplineService extends Service {
  prefix = 'disciplines';

  create = ({name, group_id}) => this._call('create', {name, group_id});
  remove = (id) => this._call('remove', {id});
  get = (id) => this._call('get', {id});
  getList = (group_id) => this._call('get_list', {group_id});
}

class DisciplineStorage {
  constructor() {
    this.data = [{
      id: 0,
      name: 'Math',
      groupId: 0,
    }, {
      id: 1,
      name: 'Physics',
      groupId: 0,
    }, {
      id: 2,
      name: 'Software engineering',
      groupId: 0,
    }, {
      id: 3,
      name: 'Develop',
      groupId: 0,
    }];
  }


  async create({title, text}) {
    this.data.unshift({title, text});
    return this.data.length;
  }

  async getList() {
    return this.data;
  }
}

export default new DisciplineService();